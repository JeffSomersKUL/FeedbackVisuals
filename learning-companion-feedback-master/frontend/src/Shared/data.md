# Data in tabel

## Bio-ingenieurs (bi)

### Totaalscore
 - score20
 -> andere velden zijn leeg

## Biomedische wetenschappen (bw)
### Totaalscore: (totaalscore type 1)
 - score20
 - juist (som juistA, juistB, juistC)
 - fout (som foutA, foutB, foutC)
 - blanco (som blancoA, blancoB, blancoC)
### Subscores:
 - wiskunde: A (op 20)
 - chemie: B (op 10)
 - basisvaardigheden: C (op 10)


## Dierengeneeskunde (dw)

### Totaalscore: Type 1
### Subscores
 - Wiskunde: A (op 10)
 - Fysica: B (op 10)
 - Chemie: C (op 10)

 - Leren:
    - VerwerkingsstrategieÃ«n: scoreD (op 20)
    - Stuurloos leergedrag: juistD (op 20)
 - Taaksturing:
    - Zelfcontrole: foutD (op 12)
    - Uitstel: blancoD (op 20)
 - Nieuwsgierigheid: scoreE (op 20)
 - Emoties:
    - Testangst: juistE (op 20)
    - Emotionele controle: foutE (op 20)
 - Samenwerking: blancoE (op 20)
 - Extraversie: scoreF (op 20)

## Economische wetenschappen (ew)
### Totaalscore
  - score20
### Subscores
  - Academische vaardigheden: enkel scoreA (op 10)
  - Wiskundige vaardigheden: enkel scoreB (op 30)

## Farmaceutische wetenschappen (fa)
### Totaalscore: Type 1

### Subscores
  - wiskunde: A (op 20)
  - chemie: B (op 10)
  - basisvaardigheden: C (op 10)

## Handelsingenieur (hi)
### Totaalscore
  - score20
### Subscores
  - Academische vaardigheden: enkel scoreA (op 10)
  - Wiskundige vaardigheden: enkel scoreB (op 30)

## Handelswetenschappen (hw)
### Totaalscore
  - score20
### Subscores
  - Academische vaardigheden: enkel scoreA (op 20)
  - Wiskundige vaardigheden: enkel scoreB (op 20)

## Ingenieurswetenschappen architect (ia)
### Totaalscore
  - Niet aanwezig (wordt niet bepaald ?)

### Subscores
  - Basisvaardigheden wiskunde: A (op 10)
  - Volledige wiskunde (inclusief basisvaardigheden): B (op 20) (25 vragenn) (kolommen B bevatten ook basisvaardigheden, zoals het hoord)
  - Ruimtelijk inzicht: C (6 vragen, omgerekend op 10) (op 10)

### Vragen
  - Vragenreeks veld is ingevuld
  - Antwoorden zijn ingevuld (comma seperated lijst)

## Industrieel ingenieur biowetenschappen (ib)
### Totaalscore
  - Niet in de data te vinden, score20 is 0 of null (bij NA)
### Subscores
  - Wiskunde: A (op 10) (blanco veld is leeg -> berekenen via overschot ?)
  - Analyse van een tekst: B (op 9) (blanco veld ook leeg)

## Industrieel ingenieur (in)
### Totaalscore
  - Niet in de data te vinden, score20 is 0 of null (bij NA)
### Subscores
  - Wiskunde: A (op 10) (blanco veld is leeg -> berekenen via overschot ?)
  - Analyse van een tekst: B (op 9) (blanco veld ook leeg)

## Ingenieurswetenschappen (ir)
### Totaalscore: type 1

### Subscores: geen

### Vragen
  - Vragenreeks veld is ingevuld
  - Antwoorden zijn ingevuld (comma seperated lijst)

## Logopedische en audiologische wetenschappen 
### Totaalscore: type 1

### Subscores
  - Wiskunde: A (op 20)
  - Taalkunde: B (op 10)
  - Basisvaardigheden: C (op 10)

## Wetenschappen (wb)
### Totaalscore: type 1

### Subscores
  - Wiskunde: A (op 20)
  - Chemie: B (op 10)
  - Basisvaardigheden: C (op 10)

## Wetenschappen wiskunde (ww)
### Totaalscore: type 1

### Subscores: geen