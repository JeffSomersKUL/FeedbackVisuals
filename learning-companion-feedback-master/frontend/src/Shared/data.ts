export interface FeedbackCodeInfo {
    feedbackCode?: string;
    session: Number;
    program: string;
}

export const getfeedbackCodeRegex = () => /^(\d+)([a-z][a-z]).*$/;
export const parseFeedbackCode = (feedbackCode: string) => {
    const feedbackCodeMatch = feedbackCode.match(getfeedbackCodeRegex());
    if (feedbackCodeMatch) {
        return {
            feedbackCode,
            session: Number(feedbackCodeMatch[1]),
            program: feedbackCodeMatch[2]
        } as FeedbackCodeInfo;
    }
    return undefined;
};

export interface Score {
    name: string;
    score: IjkingstoetsScore;
    peerScore?: PeerScore;
    cseScore?: CSEScore;
}

export type ToetsCode = "wb";

export interface ISubScore {
    name: string;
    maxScore: number;
    vragenIds?: number[];
}

export interface IConfig {
    toetsCode: ToetsCode;
    toetsSessie: number;
    aantalVragen: number;
    subscores: ISubScore[];
}

export type IStatisticsItemType = "lowest" | "highest" | "mean" | "median" | "scoreAmounts" | "amountOfStudents";

export interface IStatisticsItem {
    subScoreName?: string;
    type: IStatisticsItemType;
    value: number;
    arg?: number;
}

export interface ScoreAmount {
    score: number;
    amount: number;
}

export interface SubScoreStatistics {
    subScoreName: string;
    highestScore: number;
    lowestScore: number;
    meanScore: number;
    medianScore: number;
    scoreAmounts: ScoreAmount[];
}

export interface Statistics {
    amountOfParticipants: number;
    subScoreStatistics: SubScoreStatistics[];
}

export interface ProcessedData {
    passed: boolean;
    answers: AnswerData[];
    scores: Score[];
    feedbackCode: string;
    config: IConfig;
    statistics: Statistics;
    questionnaire: number;
}

export interface ParticipationResponse extends ProcessedData {
    canEdit: boolean;
}

export interface AllAnswersData {
    A: number;
    B: number;
    C: number;
    D: number;
    X: number;
}

export interface AnswerData {
    questionId: number;
    position: number;
    givenAnswer: string;
    correctAnswer?: string;
    allAnswers?: AllAnswersData;
    percentageCorrect?: number;
    percentageBlanco?: number;
}

export interface IjkingstoetsInfo {
    group: string; // reeks in dutch
    date: string;
    name: string;
    studyProgram: string;
}

export interface IjkingstoetsScore {
    score: number;
    maxScore: number;
    hasBreakdown: boolean;
    numberCorrect?: number;
    numberBlanco?: number;
    numberWrong?: number;
    groupLabel?: string;
}

export interface PeerScorePart {
    label: string;
    amount: number;
    color: string;
}

export interface CSEScorePart {
    scoreGroupLabel: string;
    scoreGroupColor: string;
    cseGroupLabel: string;
    cseGroupColor: string;
    percentage: number;
}

export interface PeerScore {
    parts: PeerScorePart[];
}

export interface CSEScore {
    parts: CSEScorePart[];
}

export interface LMHCSEPrediction {
    lowToTopPct: number;
    lowToMiddlePct: number;
    lowToBottomPct: number;
    middleToTopPct: number;
    middleToMiddlePct: number;
    middleToBottomPct: number;
    highToTopPct: number;
    highToMiddlePct: number;
    highToBottomPct: number;
}

export interface NegPosCSEPrediction {
    unknownToTopPct: number;
    unknownToMiddlePct: number;
    unknownToBottomPct: number;
    negativeToTopPct: number;
    negativeToMiddlePct: number;
    negativeToBottomPct: number;
    neutralToTopPct: number;
    neutralToMiddlePct: number;
    neutralToBottomPct: number;
    positiveToTopPct: number;
    positiveToMiddlePct: number;
    positiveToBottomPct: number;
}

export interface WeakGoodCSEPrediction {
    veryWeakToTopPct: number;
    veryWeakToMiddlePct: number;
    veryWeakToBottomPct: number;
    weakToTopPct: number;
    weakToMiddlePct: number;
    weakToBottomPct: number;
    meanToTopPct: number;
    meanToMiddlePct: number;
    meanToBottomPct: number;
    goodToTopPct: number;
    goodToMiddlePct: number;
    goodToBottomPct: number;
    veryGoodToTopPct: number;
    veryGoodToMiddlePct: number;
    veryGoodToBottomPct: number;
}

export interface Program {
    name: string;
    code: string;
    sessions: Array<{
        session: number;
        files: {
            feedbackFiles: string[]
            feedbackFilesLastChanged: string
            personalFeedbackFiles: string[]
            personalFeedbackFilesLastChanged: string
        }
    }>;
}

export const mandatoryFeedbackFileTypes = ["antwoorden", "config", "peerScores", "scores", "statistieken", "vraagAntwoorden", "vragenlijsten"];
export const feedbackFileTypes = [...mandatoryFeedbackFileTypes, "cseScores"
];
