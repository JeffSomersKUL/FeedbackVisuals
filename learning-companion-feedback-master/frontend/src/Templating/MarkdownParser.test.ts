import MarkdownParser from './MarkdownParser'
import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import { ProcessedData } from 'Shared/data';

const dataHigh: ProcessedData = {
    passed: true,
    answers: [],
    scores: [{
        name: 'totaal',
        score: {
            score: 17,
            maxScore: 20,
            hasBreakdown: false
        }    
    }],
    feedbackCode: '',
    statistics: {
        amountOfParticipants: 10,
        subScoreStatistics: []
    },
    questionnaire: 1,
    config: {
        toetsCode: "wb",
        toetsSessie: 15,
        aantalVragen: 20,
        subscores: []
    }
}

const dataLow: ProcessedData = {
    passed: true,
    answers: [],
    scores: [{
        name: 'totaal',
        score: {
            score: 8,
            maxScore: 20,
            hasBreakdown: false
        }
    }],
    feedbackCode: '',
    statistics: {
        amountOfParticipants: 10,
        subScoreStatistics: []
    },
    questionnaire: 1,
    config: {
        toetsCode: "wb",
        toetsSessie: 15,
        aantalVragen: 20,
        subscores: []
    }
}

describe('MarkdownParser', () => {
    describe('checkConditionals', () => {
        it('works for valid conditions', () => {
            const tekst: string = 
            `BEGIN1[als "a" < "b" en 1 < 2][als 'tekst' != 'tekst']
            # content1
            BEGIN2[als 'b' >= 'b']
            content2
            EINDE2
            EINDE1`

            expect(MarkdownParser.checkConditionals(tekst)).toBe(true)
        })

        it('fails for string without quotes in outer if', () => {
            const tekst: string =
                `BEGIN1[als a < "b" en 1 < 2][als 'tekst' != 'tekst']
            # content1
            BEGIN2[als 'b' >= 'b']
            content2
            EINDE2
            EINDE1`

            expect(() => MarkdownParser.checkConditionals(tekst)).toThrow('a is geen geldig getal. Gebruik quotes ("") rond strings en @ voor variabelennamen.')
        })

        it('fails for string without quotes in inner if', () => {
            const tekst: string =
                `BEGIN1[als 'a' < "b" en 1 < 2][als 'tekst' != 'tekst']
            # content1
            BEGIN2[als b >= 'b']
            content2
            EINDE2
            EINDE1`

            expect(() => MarkdownParser.checkConditionals(tekst)).toThrow('b is geen geldig getal. Gebruik quotes ("") rond strings en @ voor variabelennamen.')
        })

        it('works with variables', () => {
            const tekst: string =
                `BEGIN1[als @score(totaal) > 10]
            Meer dan 10
            EINDE1
            BEGIN1[als @score(totaal) <= 10]
            Maximaal 10
            EINDE1`

            expect(MarkdownParser.checkConditionals(tekst)).toBe(true)
        })
    })

    describe('filterConditionals', () => {
        it('shows if all conditions are valid', () => {
            const tekst: string =
                `BEGIN1[als "a" < "b" en 1 < 2][als 'tekst' = 'tekst']
# content1 EINDE1`

            expect(MarkdownParser.filterConditionals(tekst)).toBe('# content1 ')
        })

        it('shows if one condition is valid', () => {
            const tekst: string =
                `BEGIN1[als "a" < "b" en 1 > 2][als 'tekst' = 'tekst']
# content1 EINDE1`

            expect(MarkdownParser.filterConditionals(tekst)).toBe('# content1 ')
        })

        it('hides if no condition is valid', () => {
            const tekst: string =
                `BEGIN1[als "a" < "b" en 1 > 2][als 'tekst' != 'tekst']
            # content1
            EINDE1`

            expect(MarkdownParser.filterConditionals(tekst)).toBe('')
        })

        it('hides inner if no condition is valid', () => {
            const tekst: string =
                `BEGIN1[als "a" < "b" en 1 < 2][als 'tekst' != 'tekst']
# content1BEGIN2[als "a" < "b" en 1 > 2][als 'tekst' != 'tekst']
            some tekst
EINDE2 EINDE1`

            expect(MarkdownParser.filterConditionals(tekst)).toBe('# content1 ')
        })

        it('works with numeric variables', () => {
            const tekst: string =
                `BEGIN1[als @score(totaal) > 10] Meer dan 10 EINDE1
                 BEGIN1[als @score(totaal) <= 10] Maximaal 10 EINDE1`

            expect(MarkdownParser.filterConditionals(tekst, dataHigh).trim()).toBe('Meer dan 10')
            expect(MarkdownParser.filterConditionals(tekst, dataLow).trim()).toBe('Maximaal 10')
        })

        it('throws when variable does not exist', () => {
            const tekst: string =
                `BEGIN1[als @student.advies = 'positief'] Leuk, positief EINDE1
                 BEGIN1[als @student.advies != 'positief'] Oei, niet positief EINDE1`
            expect(() => MarkdownParser.filterConditionals(tekst, dataHigh)).toThrow('Onbekende variable @student.advies.')
            expect(() => MarkdownParser.filterConditionals(tekst, dataLow)).toThrow('Onbekende variable @student.advies.')
        })
    })

    describe('parseText', () => {
        it('shows simple text', () => {
            const parsed = MarkdownParser.parseText('Test this with some text.')
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            const { getByText } = render(parsed);
            expect(getByText('Test this with some text.')).toBeInTheDocument();
        });

        it('shows right text for bold text', () => {
            const parsed = MarkdownParser.parseText('Test **this** with some **bold** text.')
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            const { container, getByText } = render(parsed);
            expect(container).toMatchSnapshot()
        });

        it('shows right text for italic text', () => {
            const parsed = MarkdownParser.parseText('Test _this_ with some _italic_ text.')
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            const { container, getByText } = render(parsed);
            expect(container).toMatchSnapshot()

        });

        it('shows right text for bold in italic text', () => {
            const parsed = MarkdownParser.parseText('Test _this with some **italic**_ text.')
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            const { container, getByText } = render(parsed);
            expect(container).toMatchSnapshot()

        });

        it('shows right text for italic in bold text', () => {
            const parsed = MarkdownParser.parseText('Test **this with some _bold_** text.')
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            const { container, getByText } = render(parsed);
            expect(container).toMatchSnapshot()

        });

        it('errors with not closed bold / italic text', () => {
            expect(() => MarkdownParser.parseText('Test **this with some _bold_ text.')).toThrow('Bold stars not closed in Test **this with some _bold_ text.') 
            expect(() => MarkdownParser.parseText('Test **this with some _bold** text.')).toThrow('Italic underscore not closed in this with some _bold')        
            expect(() => MarkdownParser.parseText('Test this with some _bold_** text.')).toThrow('Bold stars not closed in ** text.')
            expect(() => MarkdownParser.parseText('Test **this with some bold_** text.')).toThrow('Italic underscore not closed in this with some bold_') 
        });


    })

    describe('parseLinks', () => {
        it('parses a simple link', () => {
            const parsed = MarkdownParser.parseLinks('hi there [Linktekst](http://link.com)', false)
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            let { container, getByText } = render(parsed);
            expect(container).toMatchSnapshot()

            const parsed2 = MarkdownParser.parseLinks('hi there [Linktekst](http://link.com)', true)
            container = render(parsed2).container;
            expect(container).toMatchSnapshot()
        })

        it('parses two simple links', () => {
            const parsed = MarkdownParser.parseLinks('hi there [Linktekst1](http://link.com) and [Linktekst2](http://link.com)', false)
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            let { container, getByText } = render(parsed);
            expect(container).toMatchSnapshot()

            const parsed2 = MarkdownParser.parseLinks('hi there [Linktekst1](http://link.com) and [Linktekst2](http://link.com)', true)
            container = render(parsed2).container;
            expect(container).toMatchSnapshot()
        })

        it('parses boldness etc of two simple link names if needed', () => {
            const parsed = MarkdownParser.parseLinks('hi there [**Linktekst1**](http://link.com) and [_Linktekst2_](http://link.com)', false)
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            let { container, getByText } = render(parsed);
            expect(container).toMatchSnapshot()

            const parsed2 = MarkdownParser.parseLinks('hi there [**Linktekst1**](http://link.com) and [_Linktekst2_](http://link.com)', true)
            container = render(parsed2).container;
            expect(container).toMatchSnapshot()
        })

        it('parses link in title', () => {
            const parsed = MarkdownParser.parse('###### hi there [**Linktekst1**](http://link.com) and [_Linktekst2_](http://link.com)')
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            let { container, getByText } = render(parsed);
            expect(container).toMatchSnapshot()
        })
    })

    describe('parseLists', () => {
        it('parses a simple list with one items', () => {
            const parsed = MarkdownParser.parse(
                `- item`, undefined)
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            let { container, getByText } = render(parsed);
            expect(container).toMatchSnapshot()
        })

        it('parses a simple list with three items', () => {
            const parsed = MarkdownParser.parse(
`- item
- item2
- item3`, undefined)
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            let { container, getByText } = render(parsed);
            expect(container).toMatchSnapshot()
        })

        it('parses two lists', () => {
            const parsed = MarkdownParser.parse(
`
Text before
- one item
Text between
- item
- item2
- item3
Text After`, undefined)
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            let { container, getByText } = render(parsed);
            expect(container).toMatchSnapshot()
        })

        it('parses a simple list with three items and bold and underscored values', () => {
            const parsed = MarkdownParser.parse(
`- **item**
- _item2_
- item3`, undefined)
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            let { container, getByText } = render(parsed);
            expect(container).toMatchSnapshot()
        })
    })

    describe('parseMultiples', () => {
        it('works with 0', () => {
            const parsed = MarkdownParser.parseMultiples('<<<0 question/questions>>>', false)
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            let { container, getByText } = render(parsed);
            expect(container).toMatchSnapshot()

            const parsed2 = MarkdownParser.parseMultiples('<<<0 question/questions>>>', true)
            container = render(parsed2).container;
            expect(container).toMatchSnapshot()
        })

        it('works with 1', () => {
            const parsed = MarkdownParser.parseMultiples('<<<1 question/questions>>>', false)
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            let { container, getByText } = render(parsed);
            expect(container).toMatchSnapshot()

            const parsed2 = MarkdownParser.parseMultiples('<<<1 question/questions>>>', true)
            container = render(parsed2).container;
            expect(container).toMatchSnapshot()
        })

        it('works with 10', () => {
            const parsed = MarkdownParser.parseMultiples('<<<10 question/questions>>>', false)
            const div = document.createElement('div');
            // renders without crashing
            ReactDOM.render(parsed, div);

            let { container, getByText } = render(parsed);
            expect(container).toMatchSnapshot()

            const parsed2 = MarkdownParser.parseMultiples('<<<10 question/questions>>>', true)
            container = render(parsed2).container;
            expect(container).toMatchSnapshot()
        })
    })
})