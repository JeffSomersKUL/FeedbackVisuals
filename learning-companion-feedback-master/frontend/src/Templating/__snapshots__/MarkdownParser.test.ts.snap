// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MarkdownParser parseLinks parses a simple link 1`] = `
<div>
  <span>
    hi there 
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="http://link.com"
      style="color: blue;"
      target="_blank"
    >
      Linktekst
    </a>
  </span>
</div>
`;

exports[`MarkdownParser parseLinks parses a simple link 2`] = `
<div>
  <span>
    hi there 
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="http://link.com"
      style="color: blue;"
      target="_blank"
    >
      Linktekst
    </a>
  </span>
</div>
`;

exports[`MarkdownParser parseLinks parses boldness etc of two simple link names if needed 1`] = `
<div>
  <span>
    hi there 
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="http://link.com"
      style="color: blue;"
      target="_blank"
    >
      **Linktekst1**
    </a>
    <span>
       and 
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="http://link.com"
        style="color: blue;"
        target="_blank"
      >
        _Linktekst2_
      </a>
    </span>
  </span>
</div>
`;

exports[`MarkdownParser parseLinks parses boldness etc of two simple link names if needed 2`] = `
<div>
  <span>
    hi there 
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="http://link.com"
      style="color: blue;"
      target="_blank"
    >
      <span>
        
        <span
          style="font-weight: bold;"
        >
          Linktekst1
        </span>
      </span>
    </a>
    <span>
       and 
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="http://link.com"
        style="color: blue;"
        target="_blank"
      >
        <span>
          
          <span
            style="font-style: italic;"
          >
            Linktekst2
          </span>
        </span>
      </a>
    </span>
  </span>
</div>
`;

exports[`MarkdownParser parseLinks parses link in title 1`] = `
<div>
  <div
    style="margin-top: 0px; margin-left: 0px; margin-right: 15px;"
  >
    <h1
      class="MuiTypography-root MuiTypography-h6 MuiTypography-colorInherit"
    >
      <span>
        hi there 
        <a
          class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
          href="http://link.com"
          style="color: blue;"
          target="_blank"
        >
          **Linktekst1**
        </a>
        <span>
           and 
          <a
            class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
            href="http://link.com"
            style="color: blue;"
            target="_blank"
          >
            _Linktekst2_
          </a>
        </span>
      </span>
    </h1>
  </div>
</div>
`;

exports[`MarkdownParser parseLinks parses two simple links 1`] = `
<div>
  <span>
    hi there 
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="http://link.com"
      style="color: blue;"
      target="_blank"
    >
      Linktekst1
    </a>
    <span>
       and 
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="http://link.com"
        style="color: blue;"
        target="_blank"
      >
        Linktekst2
      </a>
    </span>
  </span>
</div>
`;

exports[`MarkdownParser parseLinks parses two simple links 2`] = `
<div>
  <span>
    hi there 
    <a
      class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
      href="http://link.com"
      style="color: blue;"
      target="_blank"
    >
      Linktekst1
    </a>
    <span>
       and 
      <a
        class="MuiTypography-root MuiLink-root MuiLink-underlineHover MuiTypography-colorPrimary"
        href="http://link.com"
        style="color: blue;"
        target="_blank"
      >
        Linktekst2
      </a>
    </span>
  </span>
</div>
`;

exports[`MarkdownParser parseLists parses a simple list with one items 1`] = `
<div>
  <div
    style="margin-top: 0px; margin-left: 0px; margin-right: 15px;"
  >
    <ul>
      <li>
        item
      </li>
    </ul>
  </div>
</div>
`;

exports[`MarkdownParser parseLists parses a simple list with three items 1`] = `
<div>
  <div
    style="margin-top: 0px; margin-left: 0px; margin-right: 15px;"
  >
    <ul>
      <li>
        item
      </li>
      <li>
        item2
      </li>
      <li>
        item3
      </li>
    </ul>
  </div>
</div>
`;

exports[`MarkdownParser parseLists parses a simple list with three items and bold and underscored values 1`] = `
<div>
  <div
    style="margin-top: 0px; margin-left: 0px; margin-right: 15px;"
  >
    <ul>
      <li>
        <span>
          
          <span
            style="font-weight: bold;"
          >
            item
          </span>
        </span>
      </li>
      <li>
        <span>
          
          <span
            style="font-style: italic;"
          >
            item2
          </span>
        </span>
      </li>
      <li>
        item3
      </li>
    </ul>
  </div>
</div>
`;

exports[`MarkdownParser parseLists parses two lists 1`] = `
<div>
  <div
    style="margin-top: 0px; margin-left: 0px; margin-right: 15px;"
  >
    Text before
  </div>
  <div
    style="margin-top: 15px; margin-left: 15px; margin-right: 15px;"
  >
    <ul>
      <li>
        one item
      </li>
    </ul>
  </div>
  <div
    style="margin-top: 15px; margin-left: 15px; margin-right: 15px;"
  >
    Text between
  </div>
  <div
    style="margin-top: 15px; margin-left: 15px; margin-right: 15px;"
  >
    <ul>
      <li>
        item
      </li>
      <li>
        item2
      </li>
      <li>
        item3
      </li>
    </ul>
  </div>
  <div
    style="margin-top: 15px; margin-left: 15px; margin-right: 15px;"
  >
    Text After
  </div>
</div>
`;

exports[`MarkdownParser parseMultiples works with 0 1`] = `
<div>
  <span>
    <span
      class="MuiTypography-root MuiTypography-body1"
    >
      0 questions
    </span>
  </span>
</div>
`;

exports[`MarkdownParser parseMultiples works with 0 2`] = `
<div>
  <span>
    <span
      class="MuiTypography-root MuiTypography-body1"
    >
      0 questions
    </span>
  </span>
</div>
`;

exports[`MarkdownParser parseMultiples works with 1 1`] = `
<div>
  <span>
    <span
      class="MuiTypography-root MuiTypography-body1"
    >
      1 question
    </span>
  </span>
</div>
`;

exports[`MarkdownParser parseMultiples works with 1 2`] = `
<div>
  <span>
    <span
      class="MuiTypography-root MuiTypography-body1"
    >
      1 question
    </span>
  </span>
</div>
`;

exports[`MarkdownParser parseMultiples works with 10 1`] = `
<div>
  <span>
    <span
      class="MuiTypography-root MuiTypography-body1"
    >
      10 questions
    </span>
  </span>
</div>
`;

exports[`MarkdownParser parseMultiples works with 10 2`] = `
<div>
  <span>
    <span
      class="MuiTypography-root MuiTypography-body1"
    >
      10 questions
    </span>
  </span>
</div>
`;

exports[`MarkdownParser parseText shows right text for bold in italic text 1`] = `
<div>
  <span>
    Test 
    <span
      style="font-style: italic;"
    >
      <span>
        this with some 
        <span
          style="font-weight: bold;"
        >
          italic
        </span>
      </span>
    </span>
     text.
  </span>
</div>
`;

exports[`MarkdownParser parseText shows right text for bold text 1`] = `
<div>
  <span>
    Test 
    <span
      style="font-weight: bold;"
    >
      this
    </span>
    <span>
       with some 
      <span
        style="font-weight: bold;"
      >
        bold
      </span>
       text.
    </span>
  </span>
</div>
`;

exports[`MarkdownParser parseText shows right text for italic in bold text 1`] = `
<div>
  <span>
    Test 
    <span
      style="font-weight: bold;"
    >
      <span>
        this with some 
        <span
          style="font-style: italic;"
        >
          bold
        </span>
      </span>
    </span>
     text.
  </span>
</div>
`;

exports[`MarkdownParser parseText shows right text for italic text 1`] = `
<div>
  <span>
    Test 
    <span
      style="font-style: italic;"
    >
      this
    </span>
    <span>
       with some 
      <span
        style="font-style: italic;"
      >
        italic
      </span>
       text.
    </span>
  </span>
</div>
`;
